### PR TITLE
Improve code splitting

### PR DESCRIPTION
**Pull request by sainak**
## Proposed Changes

- update the manual chunks function to bundle required dependencies with the lazy imported components
- markdown banner on login is pre-rendered on build
- locales are lazily loaded based on usage
- huge components like PdfViewer, ExcelFIleReader, Charts, KanbanBoards are now lazy loaded
- removed envrc and fixed env handling in vite config
- fixed livekit component not splitting properly https://github.com/ohcnetwork/care_livekit_fe/pull/2

Before:
image

After:
image

image

markdown test:
image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
	- Introduced lazy loading for various components, enhancing loading performance and user experience.
	- Added support for dynamic loading of language resources via HTTP requests for internationalization.
	- Implemented a new loading indicator for components during data fetching.

- **Bug Fixes**
	- Standardized padding across multiple components for improved layout consistency.

- **Chores**
	- Removed unused files and dependencies to streamline the project structure. 

- **Documentation**
	- Updated internationalization setup documentation to reflect changes in language resource loading.

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]

**Review by shivankacker — APPROVED** [review]

**Comment by nihal467** [comment]
image

- [x] When the chart component in any section of the patient consultation page is opened for the first time, the entire page reloads. Once it's cached, it operates smoothly in subsequent uses.

To replicate, perform a hard refresh on the patient consultation page, click on the "Vital" section, and then begin selecting other sections.

Similar behavior is noted for most of the components you changed

@sainak  is this behavior expected in your recent changes?

CC: @rithviknishad

**Comment by sainak** [comment]
wrapping the components in suspens should fix it, I'll work on it

**Comment by nihal467** [comment]
@sainak once you make the PR is ready for re-review, change the label once again

File: src/Routers/SessionRouter.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
-import { Login, ResetPassword } from "@/components/Auth";
+import { lazy } from "react";
 import { useRoutes } from "raviger";
 import SessionExpired from "@/components/ErrorPages/SessionExpired";
 import InvalidReset from "@/components/ErrorPages/InvalidReset";
-import LicensesPage from "@/components/Licenses/LicensesPage";
+import ResetPassword from "@/components/Auth/ResetPassword";
+import Login from "@/components/Auth/Login";
+
+const LicensesPage = lazy(() => import("@/components/Licenses/LicensesPage"));
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/Routers/SessionRouter.tsx]: the license page opens in a new page, it is correctly handeled by the top level suspense

File: src/components/Auth/Login.tsx [hunk 2]
```diff
@@ -25,8 +24,8 @@ export const Login = (props: { forgot?: boolean }) => {
     stateLogo,
     customLogo,
     customLogoAlt,
-    customDescription,
   } = careConfig;
+  const customDescriptionHtml = __CUSTOM_DESCRIPTION_HTML__;
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/components/Auth/Login.tsx:32]: __CUSTOM_DESCRIPTION_HTML__ is set from the build process and is already sanitized

File: src/components/Common/FilePreviewDialog.tsx [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+  useState,
+  lazy,
+  Suspense,
+} from "react";
 import CareIcon, { IconName } from "../../CAREUI/icons/CareIcon";
 import ButtonV2, { Cancel } from "./components/ButtonV2";
 import DialogModal from "./Dialog";
-import PDFViewer from "./PDFViewer";
+
+const PDFViewer = lazy(() => import("@/components/Common/PDFViewer"));
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/components/Common/FilePreviewDialog.tsx:18]: invalid; the PDFViewer componenet is already lazy loaded which includes its dependencies, also the pdf.worker.min.mjs is copied during build

**Comment by sainak** [comment]
@coderabbitai review

**Comment by sainak** [comment]
>     * [x]  When the chart component in any section of the patient consultation page is opened for the first time, the entire page reloads. Once it's cached, it operates smoothly in subsequent uses.

@nihal467 now it only shows loading state for the specific component

image

File: src/i18n.ts [hunk 4]
```diff
@@ -18,21 +12,17 @@ export const LANGUAGES: { [key: string]: any } = {
...
-
 i18n
+  .use(HttpApi)
   .use(initReactI18next)
   .use(LanguageDetector)
   .init({
-    resources,
+    backend: {
+      loadPath: "/locale/{{lng}}.json",
+    },
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/i18n.ts:22]: the app is expected to be deployed on a root path

**Comment by sainak** [comment]
@coderabbitai full review

**Comment by bodhish** [comment]
@sainak nit picks are also good.

**Comment by sainak** [comment]
> @sainak nit picks are also good.

@bodhish I resolved other nits but, left these ones to keep conflicts with https://github.com/ohcnetwork/care_fe/pull/8781 to the minimum

**Comment by nihal467** [comment]
LGTM

**Pull request merged**